src/getrepos.ts

```diff
@@ -1,31 +1,36 @@
-const JupiterOneClient = require('@jupiterone/jupiterone-client-nodejs');
-const fs = require('fs');
+import { getClient } from "./getClient";
+import { readdirSync } from 'fs';
 
-const reposPath = './repos';
-const repos = fs.readdirSync(reposPath);
 const repoMap = new Map();
 
-export async function getRepoIds() {
-  const j1Client = await new JupiterOneClient({
-    account: 'j1dev',
-    username: 'j1dev',
-    password: 'dev',
-    poolId: process.env.J1_USER_POOL_ID,
-    clientId: process.env.J1_CLIENT_ID,
-    accessToken: process.env.J1_API_TOKEN,
-    dev: true,
-  }).init();
+export async function getRepoIds(repoPath: string, clientInput: {account, username, password, accessToken}) {
+  const repos = readdirSync(repoPath);
+  const j1Client = await getClient(clientInput);
 
   for (const repo of repos) {
     const repoID = await j1Client.queryV1(
       `Find CodeRepo with name='${repo}'`,
     );
     if (repoID.length === 0) {
-      console.log('Could not query Repo (' + repo + ').');
+      const newRepoPath = repoPath.substring(repoPath.length) === '/' ? repoPath + repo : repoPath + '/' + repo;
```

Comment:
`repoPath.substring(repoPath.length)` will always be an empty string.  If you're trying to get the last character you'll need to use `repoPath.length - 1` and you can also just index at the string like you would an array. E.G. `repoPath[repoPath.length - 1]`.

Reply:
Oh yeah thanks